src/platform/telemetry/telemetry.ts

```diff
@@ -5,23 +5,23 @@
 import cloneDeep = require('lodash/cloneDeep');
 import { Uri } from 'vscode';
 import { Resource } from '../common/types';
-import { IInterpreterPackages, ResourceSpecificTelemetryProperties } from '../../telemetry';
+import { ResourceSpecificTelemetryProperties } from '../../telemetry';
 import { getTelemetrySafeHashedString } from './helpers';
 import { PythonEnvironment } from '../pythonEnvironments/info';
 import { createDeferred } from '../common/utils/async';
 import { getResourceType } from '../common/utils';
-import { IServiceContainer } from '../ioc/types';
 import { getComparisonKey } from '../vscode-path/resources';
+import { traceError } from '../logging';
 
 type Context = {
     previouslySelectedKernelConnectionId: string;
 };
 export const trackedInfo = new Map<string, [ResourceSpecificTelemetryProperties, Context]>();
 export const pythonEnvironmentsByHash = new Map<string, PythonEnvironment>();
-let globalContainer: IServiceContainer | undefined = undefined;
-
-export function initializeGlobals(serviceContainer: IServiceContainer) {
-    globalContainer = serviceContainer;
```

Comment:
We no longer need to inject the entire service container, just a callback is sufficient.